eth/db/chain.py

```diff
@@ -184,32 +185,36 @@ def get_block_uncles(self, uncles_hash: Hash32) -> List[BlockHeader]:
         else:
             return rlp.decode(encoded_uncles, sedes=rlp.sedes.CountableList(BlockHeader))
 
-    # TODO: This method should take a chain of headers as that's the most common use case
-    # and it'd be much faster than inserting each header individually.
-    def persist_header(self, header: BlockHeader) -> Tuple[BlockHeader, ...]:
+    def persist_header(
+            self,
+            header: Union[BlockHeader, Iterable[BlockHeader]]) -> Tuple[BlockHeader, ...]:
```

Comment:
I would update this to be `*headers: Iterable[BlockHeader]` so that it can be called as either.

- `persist_headers(single_header)`
- `persist_headers(header_a, header_b, ...)`
- `persist_headers(*headers_to_be_persisted)`

Seems simplest in terms of typing and API.